doc/sphinx/proof-engine/tactics.rst

```diff
@@ -105,6 +105,227 @@ bindings_list`` where ``bindings_list`` may be of two different forms:
+  hypotheses :g:`H1: A` and :g:`H2: B` using the pattern :g:`[H1 H2]`
+* split a hypothesis in the form :g:`A \/ B` into two
+  subgoals using the pattern :g:`[H1|H2]`.  The first subgoal will have the hypothesis
+  :g:`H1: A` and the second subgoal will have the hypothesis :g:`H2: B`.
+
+**:ref:`Example <_intropattern_conj_ex>` doesn't work for linking to the example**
+
+**Sphinx says: "WARNING: undefined label: _intropattern_conj_ex (if the link has no caption
+the label must precede a section header)"**
+
+Splitting patterns can have more than 2 names, for example :g:`[H1 H2 H3]`
+matches :g:`A /\ B /\ C`.  Patterns can be nested:
+:g:`[[Ha|Hb] H]` can be used to split :g:`(A \/ B) /\ C`.
+
+Note that there is no equivalent to intro patterns for goals.  For a goal :g:`A /\ B`,
+use the :tacn:`split` tactic to replace the current goal with subgoals :g:`A` and :g:`B`.
+For a goal :g:`A \/ B`, use :g:`apply or_introl` to replace the current goal with :g:`A`, or
+:g:`apply or_intror` to replace the current goal with :g:`B`.
+
+The other splitting patterns described below are not frequently used.
+
+These patterns can be used when the hypothesis is an equality:
+
+* :n:`->` - replaces the right-hand side of the hypothesis with the left-hand
+  side of the hypothesis in the conclusion of the goal; the hypothesis is
+  cleared; if the left-hand side of the hypothesis is a variable, it is
+  substituted everywhere in the context and the variable is removed.
+* :n:`<-` - similar to :n:`->`, but replaces the left-hand side of the hypothesis
+  with the right-hand side of the hypothesis.
+* :n:`[= @intropatterns ]` - If the product is over an equality type,
+  applies either :tacn:`injection` or :tacn:`discriminate`
+  instead of :tacn:`destruct`; if :tacn:`injection` is applicable, the patterns
+  :n:`{+, p}` are used on the hypotheses generated by :tacn:`injection`; if the
+  number of patterns is smaller than the number of hypotheses generated, the
+  pattern :n:`?` is used to complete the list.
+
+**need example for [= ... ]**
+
+* :n:`( @simple_intropattern , ... , @simple_intropattern )` - matches
+  a product over an inductive type :g:`I` with a single
+  constructor that itself has at least `n` arguments.  It performs a case
+  analysis over the hypothesis, as :n:`destruct` would, and applies the
+  patterns to the arguments of the constructor of :g:`I`.  Note that
+  that :n:`({+ p})` is equivalent to :n:`[{+ p}]`);
+* :n:`( @simple_intropattern & ... & @simple_intropattern )` - :n:`({+& p})`
+  is a shortcut for matching with
+  :n:`(p,( ... ,( ..., p ) ... ))`.  It expects the hypothesis to be a sequence of
+  right-associative binary inductive constructors such as :g:`conj` or
+  :g:`ex_intro`; for instance, a hypothesis with type
+  :g:`A /\(exists x, B /\ C /\ D)` can be introduced via pattern
+  :n:`(a & x & b & c & d)`;
+
+
+**need example for & ... &**
+
+**A /\\ (exists x, B /\\ C /\\ D) is not valid for (a & x & b & c & d)**
```

Comment:
How did you tested it? This works:

```coq
Parameters A B C D : Prop.
Goal A /\ (exists x : nat, B /\ C /\ D) -> True.
intros (a & x & b & c & d).
```

Reply:
There's no difference but you forgot to introduce the parameters in your context first, i.e. it should have been:

```
intros A x B C D (a & x & b & c & d).
```

Reply:
OK.  I didn't expect to see the `_` in this example.  The example would be simpler without that detail, but maybe that's the way it has to be.

```
Goal forall (A: Prop) (B: Prop) (C: Prop), A /\ (exists x:nat, B /\ C) -> True.
intros *.

1 subgoal
A, B, C : Prop
______________________________________(1/1)
A /\ (exists _ : nat, B /\ C) -> True

```

gives

Reply:
Here it is, tweaked so that it does not display any `_`:

```
Goal forall (A: Prop) (B: nat -> Prop) (C: Prop), A /\ (exists x:nat, B x /\ C) -> True.
intros *.
intros (a & x & b & c).
```